src/functions/routers/action_router.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  ParameterSetDefinition,
+  PossibleParameterKeys,
+} from "../../parameters/mod.ts";
+import { SlackFunction } from "../mod.ts";
+import type {
+  BlockAction,
+  BlockActionConstraint,
+  BlockActionConstraintField,
+  BlockActionConstraintObject,
+  BlockActionHandler,
+} from "./types.ts";
+
+/**
+ * Define an actions "router" and its input and output parameters for use in a Slack application. The ActionsRouter will route incoming action events to action-specific handlers.
+ * @param {SlackFunction<InputParameters, OutputParameters, RequiredInput, RequiredOutput>} func Reference to your previously-created SlackFunction, defined via DefineFunction
+ * @returns {ActionsRouter}
+ */
+export const BlockActionsRouter = <
+  InputParameters extends ParameterSetDefinition,
+  OutputParameters extends ParameterSetDefinition,
+  RequiredInput extends PossibleParameterKeys<InputParameters>,
+  RequiredOutput extends PossibleParameterKeys<OutputParameters>,
+>(
+  func: SlackFunction<
+    InputParameters,
+    OutputParameters,
+    RequiredInput,
+    RequiredOutput
+  >,
+) => {
+  const router = new ActionsRouter(func);
+
+  // deno-lint-ignore no-explicit-any
+  const exportedHandler: any = router.export();
+
+  // deno-lint-ignore no-explicit-any
+  exportedHandler.add = ((...args: any) => {
+    router.add.apply(router, args);
+
+    return exportedHandler;
+  }) as typeof router.add;
+
+  return exportedHandler as
+    & ReturnType<typeof router.export>
+    & Pick<typeof router, "add">;
+};
+
+export class ActionsRouter<
+  InputParameters extends ParameterSetDefinition,
+  OutputParameters extends ParameterSetDefinition,
+  RequiredInput extends PossibleParameterKeys<InputParameters>,
+  RequiredOutput extends PossibleParameterKeys<OutputParameters>,
+> {
+  private routes: Array<
+    [BlockActionConstraint, BlockActionHandler<typeof this.func.definition>]
+  >;
+
+  constructor(
+    private func: SlackFunction<
+      InputParameters,
+      OutputParameters,
+      RequiredInput,
+      RequiredOutput
+    >,
+  ) {
+    this.func = func;
+    this.routes = [];
+  }
+
+  /**
+   * Add an action handler for something that can match an action event.
+   * @param {BlockActionConstraint} actionConstraint A BlockActionConstraintField(i.e. a string, array of strings or regular expression) or more complex BlockActionConstraintObject to match incoming block action events. A BlockActionConstraintField parameter are matched with a block action event's `action_id` property. A BlockActionConstraintObject parameter allows to match with other block action event properties like `block_id` as well as `action_id`. If multiple properties are specified using BlockActionConstraintObject, then the event must match ALL provided BlockActionConstraintObject properties.
+   * @returns {ActionsRouter}
+   */
+  add(
```

Comment:
nit: I'm wondering if we should give some additional thought on how this method is named. Calling `BlockActionsRouter(...).add(...)` isn't too clear to me on what I'm adding. `BlockActionsRouter(...).addHandler(...)` is more clear, or alternatively maybe `BlockActionHandlerRouter(...).add(...)`? Just thinking out loud here

Reply:
I like `addHandler`. I will rename it.